photutils/psf/photometry.py

```diff
@@ -3,1131 +3,1340 @@
-
-                sources['aperture_flux'] = aperture_photometry(
-                    image, apertures, mask=mask)['aperture_sum']
-
-                # init_guesses should be the initial 3 required
-                # parameters (x, y, flux) and then concatenated with any
-                # additional sources, if there are any
-                init_guesses = QTable(names=['x_0', 'y_0', 'flux_0'],
-                                      data=[sources['xcentroid'],
-                                            sources['ycentroid'],
-                                            sources['aperture_flux']])
-
-                # Currently only needed for the finder, as group_maker and
-                # nstar return the original table with new columns, unlike
-                # finder
-                self._get_additional_columns(sources, init_guesses)
-
-        self._define_fit_param_names()
-        for p0, param in self._pars_to_set.items():
-            if p0 not in init_guesses.colnames:
-                init_guesses[p0] = (len(init_guesses)
-                                    * [getattr(self.psf_model, param).value])
-
-        if skip_group_maker:
-            star_groups = init_guesses
-        else:
-            star_groups = self.group_maker(init_guesses)
 
-        output_tab, self._residual_image = self.nstar(
-            image, star_groups, mask=mask, progress_bar=progress_bar,
-            uncertainty=uncertainty
-        )
-        star_groups = star_groups.group_by('group_id')
+            init_params = QTable()
+            init_params['id'] = np.arange(len(sources)) + 1
+            init_params[self._xinit_name] = sources['xcentroid']
+            init_params[self._yinit_name] = sources['ycentroid']
 
-        if hasattr(output_tab, 'update'):  # requires Astropy >= 5.0
-            star_groups.update(output_tab)
         else:
-            common_cols = set(star_groups.colnames).intersection(
-                output_tab.colnames)
-            for name, col in output_tab.items():
-                if name in common_cols:
-                    star_groups.replace_column(name, col, copy=True)
-                else:
-                    star_groups.add_column(col, name=name, copy=True)
+            colnames = init_params.colnames
+            if 'id' not in colnames:
+                init_params['id'] = np.arange(len(init_params)) + 1
 
-        star_groups.meta = {'version': _get_version_info()}
+            if 'group_id' in colnames:
+                # grouper is ignored if group_id is input in init_params
+                self.grouper = None
 
-        return star_groups
+        if 'local_bkg' not in init_params.colnames:
+            if self.localbkg_estimator is None:
+                local_bkg = np.zeros(len(init_params))
+            else:
+                local_bkg = self.localbkg_estimator(
+                    data, init_params[self._xinit_name],
+                    init_params[self._yinit_name], mask=mask)
```

Comment:
```suggestion
                local_bkg = self.localbkg_estimator(
                    data, masked=np.any(mask))
```

Reply:
It was in the docstring, but I also just added an explicit check that `localbkg_estimator` is a `LocalBackground` instance or `None`.  Otherwise, it raises a `ValueError`